src/use_case/search_nearby/SearchNearbyInteractor.java

```diff
@@ -0,0 +1,26 @@
+package use_case.search_nearby;
+
+import entity.Events.Event;
+
+import java.sql.Array;
+import java.util.ArrayList;
+
+public class SearchNearbyInteractor implements SearchNearbyInputBoundary{
+    SearchNearbyDataAccessInterface dataAccessObject;
+    SearchNearbyOutputBoundary presenter;
+
+    public SearchNearbyInteractor(SearchNearbyDataAccessInterface dataAccessObject, SearchNearbyOutputBoundary presenter) {
+        this.dataAccessObject = dataAccessObject;
+        this.presenter = presenter;
+    }
+    @Override
+    public void execute(SearchNearbyInputData inputData) throws Exception {
+        ArrayList<Event> result = dataAccessObject.getNearbyEvent(inputData);
+        if (result.isEmpty()) {
+            presenter.prepareFailView();
+        } else {
+            SearchNearbyOutputData outputData = new SearchNearbyOutputData(false, result);
```

Comment:
I'm not sure what is the point of the useCaseFailed parameter when you're only providing the presenter with the output data when the use case was successful.